tik/lib/CartographerKernel.cpp

```diff
@@ -164,7 +165,7 @@ namespace TraceAtlas::tik
             {
                 auto *b = cast<BasicBlock>(BB);
                 int64_t id = GetBlockID(b);
-                if (id != -1)
+                if (id >= IDState::Valid)
```

Comment:
just use 0 here. nonnegative is fine.

Reply:
I like the enum, don't you? It's more verbose about what the condition is doing

Reply:
The enum is good for values that actually have a value. You are using it here to indicate that something is valid which is a range. Its about code clarity, and this is confusing to the uninitiated.